sunpy/map/mapbase.py

```diff
@@ -2290,6 +2290,49 @@ def draw_quadrangle(self, bottom_left, *, width: (u.deg, u.pix) = None, height:
         quad = Quadrangle(anchor, width, height, **kwergs)
         axes.add_patch(quad)
         return quad
+    
+    def _calculate_contour_levels_by_area(self, levels):
```

Comment:
This will also need to be tested both as a normal unit test but also as a figure test. See https://docs.sunpy.org/en/latest/dev_guide/contents/tests.html for more details.